lib/manageiq_foreman/lib/manageiq_foreman/inventory.rb

```diff
@@ -0,0 +1,36 @@
+module ManageiqForeman
+  class Inventory
+    attr_accessor :connection
+    def initialize(connection)
+      @connection = connection
+    end
+
+    def connect
+      # NOP
+    end
+
+    def disconnect
+      # NOP
+    end
+
+    def refresh_configuration(_target = nil)
+      {
+        :hosts      => connection.all(:hosts),
+        :hostgroups => connection.denormalized_hostgroups,
+      }
+    end
+
+    def refresh_provisioning(_target = nil)
+      {
+        :operating_systems => connection.all(:operating_system_details),
+        :media             => connection.all(:media),
+        :ptables           => connection.all(:ptables),
```

Comment:
Not needed for now, but can these be collected in parallel?  I know when we did the RHEV refresh we got enormous performance gains by parallelizing the collection.